server/api/src/rsDataObjCopy.cpp

```diff
@@ -152,15 +161,49 @@ namespace {
         return destL1descInx;
     } // open_destination_data_obj
 
-    auto close_destination_data_obj(RsComm* rsComm, const int _inx) -> int
+    auto close_destination_data_obj(RsComm* _comm, const int _l1_index, const DataObjInp& _inp) -> int
     {
-        openedDataObjInp_t dataObjCloseInp{};
-        dataObjCloseInp.l1descInx = _inx;
-        dataObjCloseInp.bytesWritten = L1desc[L1desc[_inx].srcL1descInx].dataObjInfo->dataSize;
+        const std::string_view path = static_cast<const char*>(_inp.objPath);
+
+        log_api::trace("[{}:{}] - closing [{}]", __func__, __LINE__, path);
+
+        openedDataObjInp_t close_inp{};
+        close_inp.l1descInx = _l1_index;
 
-        rodsLog(LOG_DEBUG8, "[%s:%d] - closing [%s]", __FUNCTION__, __LINE__, L1desc[_inx].dataObjInp->objPath);
+        // Declare this here because we need to return an error if the source index is invalid.
+        int ec = 0;
+
+        // The L1 descriptor index is const and confirmed to be in-bounds above. Therefore, ignore linter.
```

Comment:
This isn't quite true anymore. The value is indeed `const`, but the check occurs outside of this function. Given that the function is local to this file, this is sufficient. I will update the comment.